ballerina/tests/sms-attributes-tests.bal

```diff
@@ -0,0 +1,96 @@
+
+import ballerina/test;
+
+@test:Config {
+    groups: ["sms-attributes"]
+}
+function setSMSAttributesTest() returns error? {
+    check amazonSNSClient->setSMSAttributes({
+        monthlySpendLimit: 1,
+        deliveryStatusIAMRole: testIamRole,
+        deliveryStatusSuccessSamplingRate: 5,
+        defaultSenderID: "test",
+        defaultSMSType: TRANSACTIONAL
+    });
+}
+
+@test:Config {
+    groups: ["sms-attributes"]
+}
+function setSMSAttributesWithInvalidSenderIDTest() returns error? {
+    Error? e = amazonSNSClient->setSMSAttributes({
+        monthlySpendLimit: 1,
+        deliveryStatusIAMRole: testIamRole,
+        deliveryStatusSuccessSamplingRate: 5,
+        defaultSenderID: "testSenderID", // too long
+        defaultSMSType: TRANSACTIONAL
+    });
+    test:assertTrue(e is OperationError, "OperationError expected.");
+    test:assertEquals((<OperationError>e).message(), "DefaultSenderID is invalid");
+}
+
+@test:Config {
+    groups: ["sms-attributes"]
+}
+function setSMSAttributesWithInvalidS3BucketTest() returns error? {
+    Error? e = amazonSNSClient->setSMSAttributes({
+        monthlySpendLimit: 1,
+        deliveryStatusIAMRole: testIamRole,
+        deliveryStatusSuccessSamplingRate: 5,
+        defaultSenderID: "test",
+        defaultSMSType: TRANSACTIONAL,
+        usageReportS3Bucket: "thisS3BucketDoesNotExist"
+    });
+    test:assertTrue(e is OperationError, "OperationError expected.");
+    test:assertEquals((<OperationError>e).message(), "Invalid parameter: UsageReportS3Bucket Reason: The bucket you provided does not exist");
+}
+
+@test:Config {
+    groups: ["sms-attributes"]
+}
+function setSMSAttributesWithInvalidSampleRateTest() returns error? {
+    Error? e = amazonSNSClient->setSMSAttributes({
+        monthlySpendLimit: 1,
+        deliveryStatusIAMRole: testIamRole,
+        deliveryStatusSuccessSamplingRate: 101, // max value is 100
+        defaultSenderID: "test",
+        defaultSMSType: TRANSACTIONAL
+    });
+    test:assertTrue(e is OperationError, "OperationError expected.");
+    test:assertEquals((<OperationError>e).message(), "Invalid parameter: UsageReportS3Bucket Reason: The bucket you provided does not exist");
+}
+
+@test:Config {
+    groups: ["sms-attributes"]
+}
+function getSMSAttributesTest() returns error? {
+    check amazonSNSClient->setSMSAttributes({
+        monthlySpendLimit: 1,
+        deliveryStatusIAMRole: testIamRole,
+        deliveryStatusSuccessSamplingRate: 50,
+        defaultSenderID: "test2",
+        defaultSMSType: PROMOTIONAL
+    });
+
+    SMSAttributes attributes = check amazonSNSClient->getSMSAttributes();
+    test:assertEquals(attributes.monthlySpendLimit, 1);
+    test:assertEquals(attributes.deliveryStatusIAMRole, testIamRole);
+    test:assertEquals(attributes.deliveryStatusSuccessSamplingRate, 50);
+    test:assertEquals(attributes.defaultSenderID, "test2");
+    test:assertEquals(attributes.defaultSMSType, PROMOTIONAL);
+}
```

Comment:
```suggestion
}

```

Reply:
Fixed in [020b0a4](https://github.com/ballerina-platform/module-ballerinax-aws.sns/pull/40/commits/020b0a40d31509c3e7b499cf2ea5c620bbfc9b2b)